server/controllers/admin/employees.js

```diff
@@ -0,0 +1,313 @@
+        'JOIN fonction ON employee.fonction_id = fonction.id ' +
+        'JOIN debitor ON employee.debitor_uuid = debitor.uuid ' +
+        'JOIN creditor ON employee.creditor_uuid = creditor.uuid ' +
+        'JOIN creditor_group ON creditor_group.uuid = creditor.group_uuid ' +
+        'LEFT JOIN service ON service.id = employee.service_id ' +
+      'WHERE employee.id = ? ';
+
+    return db.exec(sql2, [req.params.id]);
+  })
+  .then(function (rows) {
+
+    if (!rows.length) {
+      throw new req.codes.ERR_NOT_FOUND();
+    }
+    res.status(200).json(rows[0]);
+  })
+  .catch(next)
+  .done();
+};
+
+/**
+* This function is responsible for creating a new employee in the database
+*
+* @param {object} request The express request object
+* @param {object} response The express response object
+* @param {object} next The express middleware next object
+*
+* @example
+* // POST /employees/ : Create a new employee
+* var employees = require('categorised/employees');
+* employees.create(request, response, next);
+*/
+exports.create = function create(req, res, next) {
+
+  var employee = req.body;
+  var sql = 'INSERT INTO employee SET ?';
+
+  // ensure dates are MySQL-parseable.
+  if (employee.dob) {
+    employee.dob = new Date(employee.dob);
+  }
+  if (employee.date_embauche) {
+    employee.date_embauche = new Date(employee.date_embauche);
+  }
+
+  // execute the SQL query
+  db.exec(sql, [ employee ])
+  .then(function (row) {
+    res.status(201).json({ id: row.insertId });
+  })
+  .catch(next)
+  .done();
+};
+
+/**
+*This function is responsible for looking for employee by names or code
+*
+* @param {object} re The express request object
+* @param {object} res The express response object
+* @param {object} next The express middleware next object
+*
+* @example
+* // GET /employees/names/x
+* var employees = require('admin/employees');
+* employees.search(req, res, next);
+**/
+
+exports.search = function search(req, res, next){
+  
+  var searchOption = '', sql = '';
+  var keyValue = '%' + req.params.value + '%';
+
+  if (req.params.key === 'code'){
+    searchOption = 'LOWER(employee.code)';
+  }else if(req.params.key === 'names'){
+    searchOption = 'LOWER(CONCAT(employee.prenom, employee.name, employee.postnom))';
+  }else{
+    return next(new req.codes.ERR_BAD_VALUE());
+  }
+
+  sql = 'SELECT ' +
+    'employee.id, employee.code AS code_employee, employee.prenom, employee.name, ' +
+    'employee.postnom, employee.locked, employee.bank, employee.bank_account, ' + 
+    'creditor.uuid as creditor_uuid, creditor.group_uuid as creditor_group_uuid, ' +
+    'creditor_group.account_id FROM employee ' +
+    'JOIN creditor ON employee.creditor_uuid = creditor.uuid ' +
+    'JOIN creditor_group ON creditor_group.uuid = creditor.group_uuid ' +
+    'WHERE ' + searchOption + ' LIKE ? LIMIT 10';
+
+    db.exec(sql, [keyValue])
+    .then(function (rows){
+      if(rows.length === 0){
```

Comment:
In this case, since we are simply searching for matches to an SQL query, there is no need to throw an error for empty matches.  You can think of it this way: It isn't an invalid query, it just didn't return any results.